mm2src/coins/z_coin/storage/blockdb/mod.rs

```diff
@@ -0,0 +1,200 @@
+#[cfg(target_arch = "wasm32")] pub(crate) mod block_idb;
+
+use mm2_core::mm_ctx::MmArc;
+use std::path::Path;
+use zcash_client_backend::data_api::BlockSource;
+use zcash_client_backend::proto::compact_formats::CompactBlock;
+use zcash_primitives::consensus::BlockHeight;
+
+cfg_native!(
+    use db_common::sqlite::rusqlite::{params, Connection, NO_PARAMS};
+    use db_common::sqlite::{query_single_row, run_optimization_pragmas};
+    use protobuf::Message;
+    use mm2_err_handle::prelude::*;
+    use std::sync::{Arc, Mutex};
+    use zcash_client_sqlite::error::{SqliteClientError as ZcashClientError, SqliteClientError};
+    use zcash_client_sqlite::NoteId;
+    use zcash_client_backend::data_api::error::Error as ChainError;
+
+    struct CompactBlockRow {
+        height: BlockHeight,
+        data: Vec<u8>,
+    }
+);
+
+#[derive(Debug, Display)]
+pub enum BlockDbError {
+    #[cfg(not(target_arch = "wasm32"))]
+    SqliteError(SqliteClientError),
+    IndexedDBError(String),
```

Comment:
Shouldn't this be `#[cfg(target_arch = "wasm32")]` as always you can combine it with `SqliteError` for a common error if you think it's better.